docs/0002-events-architecture.md

```diff
@@ -0,0 +1,84 @@
+# 2. Events Architecture
+
+Date: 2024-06-13
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, in the Visual Editor, Google Analytics (GA) tracking is implemented by hardcoding event data attributes directly into the component. This approach has limitations:
+
+- It tightly couples the GA tracking logic with the Visual Editor.
+- Any changes to GA tracking requirements necessitate changes in the Visual Editor component code.
+- The need to pass extra attributes for GA event data complicates the component logic.
+
+To address these issues, we propose implementing an Event Emitter system within the Visual Editor. This will allow us to decouple the GA tracking logic from the Visual Editor component. 
+Events will be emitted when users interact with the component (e.g., clicking a button or selecting an option from a dropdown). 
+These events can then be listened to by any part of the application to trigger GA data layer pushes or other actions, without requiring changes to the Visual Editor component itself.
+
+## Decision
+
+We will implement an Event Emitter system in the Visual Editor. The component will emit events for user interactions such as button clicks and dropdown selections. 
+These events will carry necessary data attributes, and other parts of the application can listen to these events to perform actions like GA tracking.
+
+The following steps outline the implementation:
+
+1. Create an `EventEmitter` class to manage event listeners and emit events.
+2. Integrate the `EventEmitter` into the `VisualEditor` component.
+3. Emit events from the `VisualEditor` component when user interactions occur (e.g., `clickButton`, `selectOption`).
+4. Other parts of the application can register event listeners to handle these events and push data to the GA data layer as needed.
```

Comment:
Can we replace "Other parts of the application" with Publishing Apps to make it clear where the event handlers should live?